cms/auth/static/js/auth.js

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-console */
+
+import SessionManagement from 'dis-authorisation-client-js';
+
+window.authConfig = window.authConfig || {};
+
+const {
+  wagtailAdminHomePath,
+  csrfCookieName,
+  csrfHeaderName,
+  sessionRenewalOffsetSeconds,
+  authTokenRefreshUrl,
+  idTokenCookieName,
+} = window.authConfig;
+const { origin } = window.location;
+const logoutURL = `${origin}/${wagtailAdminHomePath}logout/`;
+const extendSessionURL = `${origin}/${wagtailAdminHomePath}extend-session/`;
+
+function getCookieByName(name) {
+  console.debug('[WAGTAIL] Getting cookie by name:', name);
+  const cookies = document.cookie;
+  if (!cookies) {
+    return null;
+  }
+  const cookie = cookies
+    .split(';')
+    .map((c) => c.trim())
+    .map((c) => c.split('='))
+    .find((pair) => pair[0] === name);
+  return cookie ? cookie[1] : null;
+}
+
+// Performs a fetch request with CSRF protection.
+const fetchWithCsrf = async (url, method = 'POST', body = null, additionalHeaders = {}) => {
+  const csrfToken = getCookieByName(csrfCookieName);
+  if (!csrfToken) {
+    console.error('[WAGTAIL] CSRF token not found.');
+    return null;
+  }
+
+  const headers = {
+    'Content-Type': 'application/json',
+    [csrfHeaderName]: csrfToken,
+    ...additionalHeaders,
+  };
+
+  const config = {
+    method,
+    headers,
+    credentials: 'include',
+    ...(body && { body: JSON.stringify(body) }),
+  };
+
+  try {
+    return await fetch(url, config);
+  } catch (error) {
+    console.error(`[WAGTAIL] Error during ${method} request to ${url}:`, error);
+    return null;
+  }
+};
+
+// Initiates the logout process and handles redirection.
+const logout = async () => {
+  console.info('[WAGTAIL] Initiating logout...');
+  try {
+    const response = await fetchWithCsrf(logoutURL);
+    if (response?.ok) {
+      const redirectURL = response.redirected ? response.url : origin;
+      console.info('[WAGTAIL] Logout successful. Redirecting to:', redirectURL);
+      window.location.href = redirectURL;
+    } else {
+      console.error('[WAGTAIL] Logout failed:', response);
+      // Reload the page to handle any session inconsistencies.
+      window.location.reload();
```

Comment:
I'm thinking that any developer will need to enable persistent logs to see the error in the console after reloading the window. If they don't have it enabled they might not notice it. However I do understand why you might want to reload the window.

Reply:
Yes, I am inclined to keep this as it is.